### PR TITLE
Convert unused ctor to static factory method

### PR DESCRIPTION
**Pull request by davidxia**
Make AgentProxy interface throw IOException instead of having
implementation throw it as a RuntimeException.

Fix a spelling mistake.

### CODE REVIEW

**Comment by davidxia** [comment]
@mattnworb @gimaker @rohansingh

**Comment by davidxia** [comment]
TODO
- try to find a nicer way to test than using all those bytes
- if we don't do the above, make test constants public attributes in an interface

File: src/main/java/com/spotify/sshagentproxy/DefaultAgentProxy.java [hunk 1]
```diff
@@ -61,31 +59,26 @@ public DefaultAgentProxy() {
...
-      in = new AgentInput(Channels.newInputStream(channel));
-      out = new AgentOutput(Channels.newOutputStream(channel));
+      return new DefaultAgentProxy(new AgentInput(Channels.newInputStream(channel)),
+                                   new AgentOutput(Channels.newOutputStream(channel)));
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
-  @VisibleForTesting
```
Comment (mattnworb) [inline src/main/java/com/spotify/sshagentproxy/DefaultAgentProxy.java:71]: technically the package private constructor is still only visible for testing

**Comment by mattnworb** [comment]
:+1:

**Pull request merged**